lms/util/h.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+
+"""Utility functions for working with the h API."""
+
+from __future__ import unicode_literals
+
+import random
+import string
+
+
+USERNAME_ALLOWED_CHARS = string.ascii_letters + string.digits + "._"
+"""The characters that are allowed in h usernames."""
+
+
+DISPLAY_NAME_MAX_LENGTH = 30
+"""The maximum length of an h display name."""
+
+
+USERNAME_MAX_LENGTH = 30
+"""The maximum length of an h username."""
+
+
+def generate_display_name(request_params):
+    full_name = (request_params.get("lis_person_name_full") or "").strip()
+    given_name = (request_params.get("lis_person_name_given") or "").strip()
+    family_name = (request_params.get("lis_person_name_family") or "").strip()
+
+    if full_name:
+        display_name = full_name
+    else:
+        display_name = " ".join((given_name, family_name))
+
+    display_name = display_name.strip()
+
+    display_name = display_name or "Anonymous"
```

Comment:
I think there may be some discussion to be had around whether "Anonymous" is the fallback string we want here, but I think I'm OK with letting that be a follow-on topic and not blocking this work on that decision.